### ISSUE TITLE
Automaticly Refresh token doesn't set the header 'auth'

### ISSUE THREAD

**Issue by Inza9hi** [requests/requests-oauthlib#250]
Sorry, I am newbee in Python. Maybe this is a ridiculous question.
When I get the token, I will save the access_token,refresh_token,and **expires_at** in the database.
Some time later, when the access_token expires,  the client will refresh the token like this:

_From class OAuth2Session_
` if self.auto_refresh_url:
                    log.debug('Auto refresh is set, attempting to refresh at %s.',
                              self.auto_refresh_url)
                    token = self.refresh_token(self.auto_refresh_url, **kwargs)`

But how to pass the auth to that method?

**Comment by Lukasa** [comment]
Are you sure you need the auth to refresh the token?

**Comment by Lukasa** [comment]
Either way, the answer is the same: you need to provide a dictionary of `auto_refresh_kwargs` in addition to your `auto_refresh_url` that provide any specific extra kwargs you'll need.

**Comment by Inza9hi** [comment]
of course, refreshing token needs the auth.
in the case of fitbit:
https://dev.fitbit.com/docs/oauth2/#refreshing-tokens

```
POST https://api.fitbit.com/oauth2/token
Authorization: Basic Y2xpZW50X2lkOmNsaWVudCBzZWNyZXQ=
Content-Type: application/x-www-form-urlencoded

grant_type=refresh_token&refresh_token=abcdef01234567890abcdef01234567890abcdef01234567890abcdef0123456
```

The problem is, when the token expires, I need to re-new the oauthClient, which does not have the auth in the header.

I've provided the _auto_refresh_kwargs_ and the  _auto_refresh_url_
As I quoted in the post, the code doesn't put the auth in the header.

Or don't other standard Oauth2 process need the auth?
